WordPress/gutenberg
# Enable an optional namespace parameter for hasAction & hasFilter

## Description
Core originally in from https://github.com/WordPress/packages/pull/106/
Fixes https://github.com/WordPress/gutenberg/issues/7819

## How has this been tested?
* Add a hook with its required namespace.
* Call matching `hasfilter/hasAction` and verify results correctly.
* testing calling without namespace continues to work as expected.
* See test files for various tests.

## Types of changes
* Add an optional namespace parameter for `hasfilter` & `hasAction` 
* Bdetter match core php functions`has_action` `has_filter` which include this second parameter

## Checklist:
